drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,335 @@
+
+		if (!(channel_cfg->channel_num > 0 &&
+			channel_cfg->channel_num < SOC_TOUCH_SENSOR_NUM)) {
+			LOG_ERR("Touch %d configuration failed: "
+				"Touch channel error", i);
+			return -EINVAL;
+		}
+
+#if defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+		if (channel_cfg->channel_num == SOC_TOUCH_DENOISE_CHANNEL) {
+			LOG_ERR("Touch %d configuration failed: "
+				"TOUCH0 is internal denoise channel", i);
+			return -EINVAL;
+		}
+#endif  /* defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3) */
+
+		gpio_num_t gpio_num = touch_sensor_channel_io_map[channel_cfg->channel_num];
+
+		rtc_gpio_init(gpio_num);
+		rtc_gpio_set_direction(gpio_num, RTC_GPIO_MODE_DISABLED);
+		rtc_gpio_pulldown_dis(gpio_num);
+		rtc_gpio_pullup_dis(gpio_num);
+
+		touch_hal_config(channel_cfg->channel_num);
+#if defined(CONFIG_SOC_SERIES_ESP32)
+		touch_hal_set_threshold(channel_cfg->channel_num, 0);
+		touch_hal_set_group_mask(BIT(channel_cfg->channel_num),
+					BIT(channel_cfg->channel_num));
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+		touch_hal_set_channel_mask(BIT(channel_cfg->channel_num));
+
+		channel_data->status = 0;
+#if defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+		channel_data->last_status = 0;
+#endif /* defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3) */
+		channel_data->dev = dev;
+		k_work_init_delayable(&channel_data->work,
+					touch_sensor_change_deferred);
+	}
+
+#if defined(CONFIG_SOC_SERIES_ESP32)
+	for (int i = 0; i < num_channels; i++) {
+		const struct touch_sensor_channel_config
+				*channel_cfg = &dev_cfg->channel_cfg[i];
+
+		while (!touch_hal_meas_is_done()) {
+		}
+		uint16_t touch_value = touch_hal_read_raw_data(channel_cfg->channel_num);
+
+		touch_hal_set_threshold(channel_cfg->channel_num,
+					touch_value * (100 - channel_cfg->channel_num) / 100);
+	}
+
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	touch_filter_config_t filter_info = {
+		.mode = dev_cfg->filter_mode,
+		.debounce_cnt = dev_cfg->filter_debounce_cnt,
+		.noise_thr = dev_cfg->filter_noise_thr,
+		.jitter_step = dev_cfg->filter_jitter_step,
+		.smh_lvl = dev_cfg->filter_smooth_level,
+	};
+
+	touch_hal_filter_set_config(&filter_info);
+	touch_hal_filter_enable();
+
+	touch_hal_timeout_enable();
+	touch_hal_timeout_set_threshold(SOC_TOUCH_PAD_THRESHOLD_MAX);
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+
+	/* Register touch interrupt ISR, enable intr type. */
+#if defined(CONFIG_SOC_SERIES_ESP32)
+	dev_data->rtc_intr_msk = RTC_CNTL_TOUCH_INT_ST_M;
+	rtc_isr_install(&rtc_isr, (void *)dev);
+	touch_hal_intr_enable();
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	dev_data->rtc_intr_msk = RTC_CNTL_TOUCH_DONE_INT_ST_M
+				|RTC_CNTL_TOUCH_ACTIVE_INT_ST_M
+				|RTC_CNTL_TOUCH_INACTIVE_INT_ST_M
+				|RTC_CNTL_TOUCH_SCAN_DONE_INT_ST_M
+				|RTC_CNTL_TOUCH_TIMEOUT_INT_ST_M;
```

Comment:
```c
        dev_data->rtc_intr_msk = RTC_CNTL_TOUCH_DONE_INT_ST_M |
                                 RTC_CNTL_TOUCH_ACTIVE_INT_ST_M |
                                 RTC_CNTL_TOUCH_INACTIVE_INT_ST_M |
                                 RTC_CNTL_TOUCH_SCAN_DONE_INT_ST_M |
                                 RTC_CNTL_TOUCH_TIMEOUT_INT_ST_M;
```

if you really want it here, or maybe you could have a singe define above and use it here? same below

Reply:
Better putting it under a define